<?xml version="1.0"?>
<root main_tree_to_execute="BehaviorTree">
    <!-- ////////// -->
    <BehaviorTree ID="BehaviorTree">
        <Sequence>
            <Action ID="MoveBase" goal="dolly_0"/>
            <Sequence>
                <Action ID="FindDolly"/>
                <Action ID="AutoDockAction"/>
                <Action ID="StopDolly"/>
            </Sequence>
        </Sequence>
    </BehaviorTree>
    <!-- ////////// -->
    <TreeNodesModel>
        <Action ID="AutoDockAction"/>
        <Condition ID="BatteryCheck"/>
        <Action ID="FindDolly"/>
        <Condition ID="IsBatteryCharged"/>
        <Action ID="MoveBase">
            <input_port name="goal"/>
        </Action>
        <Action ID="StopDolly"/>
    </TreeNodesModel>
    <!-- ////////// -->
</root>
